File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-flutter/composed/lib/src/realtime_module.dart
import 'package:backend_sdk/backend_sdk.dart';

import 'context.dart';
import 'types.dart';

class CrawChatRealtimeModule {
  final CrawChatSdkContext context;

  CrawChatRealtimeModule(this.context);

  Future<RealtimeSubscriptionSnapshot?> replaceSubscriptions(
    SyncRealtimeSubscriptionsRequest body,
  ) {
    return context.backendClient.realtime.syncRealtimeSubscriptions(body);
  }

  Future<RealtimeEventWindow?> pullEvents([CrawChatQueryParams? params]) {
    return context.backendClient.realtime.listRealtimeEvents(params);
  }

  Future<RealtimeAckState?> ackEvents(AckRealtimeEventsRequest body) {
    return context.backendClient.realtime.ackRealtimeEvents(body);
  }
}

File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-typescript/composed/src/realtime-module.ts
import type {
  AckRealtimeEventsRequest,
  QueryParams,
  RealtimeAckState,
  RealtimeEventWindow,
  RealtimeSubscriptionSnapshot,
  SyncRealtimeSubscriptionsRequest,
} from './types.js';
import type { CrawChatSdkContext } from './sdk-context.js';

export class CrawChatRealtimeModule {
  constructor(private readonly context: CrawChatSdkContext) {}

  replaceSubscriptions(
    body: SyncRealtimeSubscriptionsRequest,
  ): Promise<RealtimeSubscriptionSnapshot> {
    return this.context.backendClient.realtime.syncRealtimeSubscriptions(body);
  }

  pullEvents(params?: QueryParams): Promise<RealtimeEventWindow> {
    return this.context.backendClient.realtime.listRealtimeEvents(params);
  }

  ackEvents(body: AckRealtimeEventsRequest): Promise<RealtimeAckState> {
    return this.context.backendClient.realtime.ackRealtimeEvents(body);
  }
}
